numpy/_core/src/multiarray/convert_datatype.h

```diff
@@ -12,7 +12,8 @@ extern NPY_NO_EXPORT npy_intp REQUIRED_STR_LEN[];
 #define NPY_USE_LEGACY_PROMOTION 0
 #define NPY_USE_WEAK_PROMOTION 1
 #define NPY_USE_WEAK_PROMOTION_AND_WARN 2
-extern NPY_NO_EXPORT int npy_promotion_state;
+
+extern NPY_NO_EXPORT NPY_TLS int npy_promotion_state;
```

Comment:
```suggestion
```
Since you add the functions, keeping the definition strange?

Unless you go back to using it directly (if that is even technically possible).